src/vcHTTP.cpp

```diff
@@ -0,0 +1,318 @@
+
+      char *pColonStart = pColon + 1;
+
+      while ((*pColonStart == ' ' || *pColonStart == '\t') && pColonStart < pLineEnd)
+        ++pColonStart;
+
+      pHeaders->tokens.Set("%s = '%s'", pLineStart, pColonStart);
+
+      if (udStrEqual(pLineStart, "User-Agent"))
+        pHeaders->pUserAgentString = pHeaders->tokens.Get("User-Agent").AsString(); //The string is owned by the value
+
+      //Remove null terminators
+      *pLineEnd = endToken;
+      *pColon = ':';
+    }
+
+    //Remove extra whitespace
+    int total = 0;
+    while (*pCarat == '\r' || *pCarat == '\n')
+    {
+      ++pCarat;
+      ++total;
+    }
+
+    //Exit if the body's been reached, signified by "\r\n\r\n", or if the end of the header has been reached and there's no body, signified by '\0'.
+    if (*pCarat == '\0' || total == 4)
+      break;
+  } while (true);
+
+  //Copy body pointer out
+  if (ppBodyData != nullptr && *pCarat != '\0')
+    *ppBodyData = pCarat;
+
+  //Handle the Cookies strings
+  if (pHeaders->tokens.Get("Cookie").IsString())
+  {
+    udJSON *pCookieValue;
+    pHeaders->tokens.Get(&pCookieValue, "Cookie");
+
+    char *pCookieStr;
+    pCookieValue->ExtractAndVoid((const char**)&pCookieStr);
+
+    vcHTTP_SplitAndInject(&pHeaders->cookies, pCookieStr, ';', '=', true);
+
+    udFree(pCookieStr);
+  }
+
+  if (!pHeaders->tokens.Get("Cookie").IsObject())
+    pHeaders->tokens.Set("Cookie = {}");
+}
+
+const char* vcHTTP_StatusAsText(vcHTTPStatus status)
+{
+  switch (status)
+  {
+  case vcHTTP_OK:
+    return "OK";
+  case vcHTTP_Created:
+    return "Created";
+  case vcHTTP_Accepted:
+    return "Accepted";
+  case vcHTTP_NoContent:
+    return "No Content";
+  case vcHTTP_PartialContent:
+    return "Partial Content";
+  case vcHTTP_Moved:
+    return "Moved Permanently";
+  case vcHTTP_Redirect:
+    return "Found";
+  case vcHTTP_NotModified:
+    return "Not Modified";
+  case vcHTTP_BadRequest:
+    return "Bad Request";
+  case vcHTTP_Forbidden:
+    return "Forbidden";
+  case vcHTTP_NotFound:
+    return "Not Found";
+  case vcHTTP_MethodNotAllowed:
+    return "Method Not Allowed";
+  case vcHTTP_RequestTimeout:
+    return "Request Time - out";
+  case vcHTTP_Conflict:
+    return "Conflict";
+  case vcHTTP_LengthRequired:
+    return "Length Required";
+  case vcHTTP_PreconditionFailed:
+    return "Precondition Failed";
+  case vcHTTP_PayloadTooLarge:
+    return "Request Entity Too Large";
+  case vcHTTP_URITooLong:
+    return "Request - URI Too Large";
+  case vcHTTP_UnsupportedFormat:
+    return "Unsupported Media Type";
+  case vcHTTP_RangeNotSatisfiable:
+    return "Requested range not satisfiable";
+  case vcHTTP_ServerError:
+    return "Internal Server Error";
+  case vcHTTP_NotImplemented:
+    return "Not Implemented";
+  case vcHTTP_ServiceUnavailable:
+    return "Service Unavailable";
+  case vcHTTP_BadHTTPVersion:
+    return "HTTP Version not supported";
+  default:
+    return "Internal Server Error";
```

Comment:
We shouldn't have a `default` (new entries in the `vcHTTP` enum should be handled correctly) and the fallback string shouldn't be an internal server error - only the 5xx range are server errors. A better fallback would be "unknown HTTP status code" or something.